chainntnfs/btcdnotify/btcd.go

```diff
@@ -585,34 +594,49 @@ func (b *BtcdNotifier) confDetailsManually(txid *chainhash.Hash,
 	return nil, nil
 }
 
-// handleBlocksConnected applies a chain update for a new block. Any watched
-// transactions included this block will processed to either send notifications
-// now or after numConfirmations confs.
+// handleBlocksConnected applies a chain update for a new block. This will
+// dispatch notifications requested by subscribers.
+//
 // TODO(halseth): this is reusing the neutrino notifier implementation, unify
 // them.
 func (b *BtcdNotifier) handleBlockConnected(newBlock *filteredBlock) error {
 	// First we'll notify any subscribed clients of the block.
 	b.notifyBlockEpochs(int32(newBlock.height), &newBlock.hash)
 
+	// Then, we'll update the spend height hint for all of our watched
+	// outpoints. This is safe to do as we do not watch already spent
+	// outpoints for spend notifications.
+	ops := make([]wire.OutPoint, 0, len(b.spendNotifications))
+	for op := range b.spendNotifications {
+		ops = append(ops, op)
+	}
+
+	if len(ops) > 0 {
+		err := b.hintCache.CommitSpendHint(newBlock.height, ops...)
```

Comment:
Seems like it would make more sense to do this after the block is inspected. At this point I guess the database can actually be inconsistent.

Reply:
Fixed.